include/dr/detail/ranges.hpp

```diff
@@ -152,8 +152,10 @@ struct local_fn_ {
              std::contiguous_iterator<rng::iterator_t<R>> ||
              rng::contiguous_range<R>)
   auto operator()(R &&r) const {
-    if constexpr (has_local_method<rng::iterator_t<R>>) {
-      return std::span(rng::begin(r).local(), rng::size(r));
+    if constexpr (has_local_method<R>) {
```

Comment:
this operator is having 5 requires, but 4 _if_ branches. rng::contiguous_range<R> is missing its _if_

I think pattern with separated operators is more readable than repeated _if constexpr_ and also immune for missing _if_ bug.
```
template <class R> requires(rng::forward_range<R> && has_local_adl<R>)
auto operator()(R &&r) const { return r.local(); }

template <class R> requires(rng::forward_range<R> && has_local_method<rng::iterator_t<R>>)
auto operator()(R &&r) const { return ...; }

template <rng::contiguous_range<R>>
auto operator()(R &&r) const { return ...;}

...
```
This is the pattern I see in ranges-v3 library - many operator() methods without _if_-s inside. E.g. in [all view](https://github.com/ericniebler/range-v3/blob/master/include/range/v3/view/all.hpp#L70) or [cartesian_product view](https://github.com/ericniebler/range-v3/blob/master/include/range/v3/view/cartesian_product.hpp#L461)

Reply:
We may always make them mutually exclusive. I understand compiler will catch such a cases and force programmer to think what he/she wants instead of silently rely on order inside `if ... else if ... else ...`.

But if we don't have strong opinion/agreement about what is definitely better let's don't do such refactors.